cpp/include/cudf/io/functions.hpp

```diff
@@ -238,6 +239,75 @@ table_with_metadata read_csv(
     read_csv_args const& args,
     rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+
+/**
+ * @brief Settings to use for `write_csv()`
+ */
+struct write_csv_args: private detail::csv::writer_options { 
```

Comment:
Is there a specific reason to use (private) inheritance instead of composition here?

Reply:
Yes, encapsulation. 'public' inheritance creates the strongest coupling (short of `friend`-ship) between classes. Also, it implements `IS-A` idiom. `*_args` are really _not_ `*_options`. 

`private` inheritance implements `IS-IMPLEMENTED-IN-TERMS-OF` idiom. Which actually is the case here.

Reply:
An alternative way to get implicit slicing while using composition could be to use a user-defined conversion operator like:
```c++
operator detail::csv::writer_options& () {
  return options_;
}
```

Reply:
True, but trampolining calls through the member is still there, if composition is used. Also, I can quickly expose the private base class' methods with `using`:
```
class args: private options {
  using options::foo; //etc.
...
};
```